src/coreclr/jit/gentree.cpp

```diff
@@ -6957,8 +6957,9 @@ GenTreeIntCon* Compiler::gtNewStringLiteralLength(GenTreeStrCon* node)
         return gtNewIconNode(0);
     }
 
-    int             length = -1;
-    const char16_t* str    = info.compCompHnd->getStringLiteral(node->gtScpHnd, node->gtSconCPX, &length);
+    constexpr int maxStrSize      = 1024;
+    char16_t      str[maxStrSize] = {};
```

Comment:
Does `= {}` cause the buffer to be zero initialized? That is writing 2kB of zeros...

Reply:
Yes, it's not needed